pkg/querier/querier.go

```diff
@@ -92,6 +95,11 @@ func (q *Querier) forAllIngesters(f func(logproto.QuerierClient) (interface{}, e
 
 // Query does the heavy lifting for an actual query.
 func (q *Querier) Query(ctx context.Context, req *logproto.QueryRequest) (*logproto.QueryResponse, error) {
+	oldestStartTime := time.Now().Add(-q.cfg.MaxLookBackPeriod)
```

Comment:
This means queries are broken for the default value (0). Have a check for `MaxLookBackPeriod > 0` :)